woocommerce/woocommerce-gateway-stripe
# Add end point to GET/POST Stripe account keys

# Changes proposed in this Pull Request:
Fixes https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1834

# Description
Add a REST end point to get/add/update Stripe account settings. Include validations similar to the existing ones here: https://github.com/woocommerce/woocommerce-gateway-stripe/blob/3543449b5c06a319a38d441ca6d7803e6a54d4d2/includes/class-wc-gateway-stripe.php#L1227-L1232. 

# Testing instructions
### Testing GET
1. Make sure `_wcstripe_feature_upe_settings` flag is enabled.
2. Make a `GET` API request to `https://localhost/wp-json/wc/v3/wc_stripe/account_keys` I used postman with `Basic Auth`, `Username` as consumer key, and `Password` as consumer secret from https://woocommerce.github.io/woocommerce-rest-api-docs/?php#authentication
3. You should see it responds with the following:
```
{
    "settings": {
        "test_publishable_key": "pk_test_12312313",
        "test_secret_key": "sk_test_9781263817263",
        "publishable_key": "",
        "secret_key": "sk_live_test",
        "test_webhook_secret": null,
        "webhook_secret": null
    }
}
```

### Testing POST
1. Make a `POST` request to `https://localhost/wp-json/wc/v3/wc_stripe/account_keys` with the following payload, feel free to use different keys, refer to the keys above in the response from "Testing GET":
```
{
    "publishable_key": "",
    "secret_key": "sk_live_test"
}
```
2. You should be able to update keys and see the following:
```
{
    "result": "success"
}
```
